README.md

```diff
@@ -1,108 +1,66 @@
 
 # Quick Start Guide
-Each service has to be launched on a separate Terminal instance. All commands start from the project root directory.
+Our development setup revolves around Docker, where all backend services are dockerised in a single `docker-compose`. This setup allows for hot-reloading as well, as the containers are setup with volumes that point to the respective directories. This means that if a file changes in the directory, it will be reflected within the container as they are pointing to the same files. As such, we have to do the necessary setups for each service before running the containers. This setup could include installing dependencies, setting up the database clients, etc. For this, we have defined convenience scripts for the ease of setup.
+
+The frontend does not support containerisation due to certain quirks with Next.js 13, thus you can run it standalone in your terminal.
 
-## Install all dependencies and setup Husky
+## Setup root dependencies
+The root `package.json` defines some convenience scripts that we will be using during development. To set them up, we have to install the root dependencies first.
 ```
-pnpm i
+npm install
 ```
 
-## User Service
-1. Navigate to `user-service`.
-   
-   ```
-   cd services/user-service
-   ```
-2. Create enviroment variables.
-    ```
-    echo "DATABASE_URL=postgres://<username>:<password>@localhost:<port>/<db-name>?schema=public" >> .env
-    echo "PORT=<user-service-port>" >> .env
-    ```
-3. Install dependencies.
-    ```
-    npm ci
-    ```
-4. Run databases on docker
-    ```
-    docker compose up --build -d
-    ```
+## Setup all services
+We automate `npm i` in all services through a convenience script that can be used in the root directory. This runs `npm i` in all of the microservices concurrently, and runs post-scripts i.e. `postinstall` in all of them as well to setup the databases, etc.
+```
+npm run install:services
+```
 
-5. Run microservice.
-   ```
-   npm run dev
-   ```
+## Start all services (other than frontend)
+The following command requires Docker (we recommend installing [Docker Desktop](https://www.docker.com/products/docker-desktop/)). This command will spin up the containers defined in the `docker-compose.dev.yml`
+```
+npm run dev:start
+```
 
-## Question Service
-1. Navigate to `question-service`.
+## Frontend
+1. Navigate to `services/frontend`.
    
    ```
-   cd services/question-service
+   cd frontend
    ```
-2. Install dependencies.
+2. Create a `.env` with the following attributes
     ```
-    npm ci
+   NEXT_PUBLIC_SERVICE_USER_URL=localhost:2001
+   NEXT_PUBLIC_SERVICE_QUESTION_URL=localhost:5001
+   NEXT_PUBLIC_SERVICE_HISTORY_URL=localhost:1001
+   NEXT_PUBLIC_GITHUB_ClIENT_ID=/** fill this in */
+   NEXT_PUBLIC_GITHUB_CLIENT_SECRET=/** fill this in */
+   NEXTAUTH_URL=http://localhost:3000
+   NEXTAUTH_SECRET=<FILL IN>
```

Comment:
very small small nit, can ignore if you want

Reply:
Resolved! thanks for spotting that out jon!